moveit_ros/move_group/src/default_capabilities/query_planners_service_capability.cpp

```diff
@@ -49,118 +50,153 @@ void MoveGroupQueryPlannersService::initialize()
+void MoveGroupQueryPlannersService::queryInterface(
+    const std::shared_ptr<moveit_msgs::srv::QueryPlannerInterfaces::Request>& /* unused */,
     const std::shared_ptr<moveit_msgs::srv::QueryPlannerInterfaces::Response>& res)
 {
   for (const auto& planning_pipelines : context_->moveit_cpp_->getPlanningPipelines())
   {
-    const auto& pipeline_id = planning_pipelines.first;
     const auto& planning_pipeline = planning_pipelines.second;
-    const planning_interface::PlannerManagerPtr& planner_interface = planning_pipeline->getPlannerManager();
-    if (planner_interface)
+
+    // TODO(sjahr): Update for multiple planner plugins
+    const auto& planner_plugin_names = planning_pipeline->getPlannerPluginNames();
+    if (planner_plugin_names.empty())
+    {
+      RCLCPP_ERROR(moveit::getLogger(), "Pipeline '%s' does not have any planner plugins.",
+                   planning_pipelines.first.c_str());
+      return;
+    }
+    const auto planner_interface = planning_pipeline->getPlannerManager(planner_plugin_names.at(0));
+    if (!planner_interface)
     {
-      std::vector<std::string> algs;
-      planner_interface->getPlanningAlgorithms(algs);
-      moveit_msgs::msg::PlannerInterfaceDescription pi_desc;
-      pi_desc.name = planner_interface->getDescription();
-      pi_desc.pipeline_id = pipeline_id;
-      planner_interface->getPlanningAlgorithms(pi_desc.planner_ids);
-      res->planner_interfaces.push_back(pi_desc);
+      RCLCPP_ERROR(moveit::getLogger(), "Requesting planner '%s' from '%s' returned a null pointer.",
+                   planner_plugin_names.at(0).c_str(), planning_pipelines.first.c_str());
     }
+    std::vector<std::string> algs;
+    planner_interface->getPlanningAlgorithms(algs);
+    moveit_msgs::msg::PlannerInterfaceDescription pi_desc;
+    pi_desc.name = planner_interface->getDescription();
+    pi_desc.pipeline_id = planning_pipelines.first;
+    planner_interface->getPlanningAlgorithms(pi_desc.planner_ids);
+    res->planner_interfaces.push_back(pi_desc);
   }
-  return true;
 }
 
-bool MoveGroupQueryPlannersService::getParams(const std::shared_ptr<rmw_request_id_t>& /* unused */,
-                                              const std::shared_ptr<moveit_msgs::srv::GetPlannerParams::Request>& req,
+void MoveGroupQueryPlannersService::getParams(const std::shared_ptr<moveit_msgs::srv::GetPlannerParams::Request>& req,
                                               const std::shared_ptr<moveit_msgs::srv::GetPlannerParams::Response>& res)
 {
   const planning_pipeline::PlanningPipelinePtr planning_pipeline = resolvePlanningPipeline(req->pipeline_id);
   if (!planning_pipeline)
-    return false;
+  {
+    RCLCPP_ERROR(moveit::getLogger(), "Pipeline '%s' does not exist.", req->pipeline_id.c_str());
+    return;
+  }
 
-  const planning_interface::PlannerManagerPtr& planner_interface = planning_pipeline->getPlannerManager();
-  if (planner_interface)
+  // TODO(sjahr): Update for multiple planner plugins
```

Comment:
Does this need to get done, or is it updated?

Reply:
This needs to be updated. For now the first planner in the chain is used which is usually OMPL. Updating this involves more changes in the visualization with RVIZ and the moveit plugins which is why I'd like to keep this out of the PR